### DISCUSSION TITLE
Possibility of reviving CBT1/2/3?

### DISCUSSION THREAD

**Discussion by SMNFXCN** [Melledy/Grasscutter#1319] [Q&A]
Title.
Documenting as well as being able to play pre-release versions of Genshin would be interesting.
Even if it's not a high priority would it be done someday?

**Comment by 4Benj** [comment]
As far as I know, definitely not with CBT1, idk about other releases.

**Reply by SMNFXCN** [reply]
Curious about why not CBT1 specifically?

**Reply by 4Benj** [reply]
iirc CBT1 network protocol and the way packets are sent and received are entirely different.
Note: This is not based on my research but on the research of another who shared their results in the GC discord server.

**Reply by Dasutein** [reply]
Doesn't [Lumie](https://twitter.com/lumie_lumie/status/1371512105606733826) have a private server for CBT1? I thought that was how they were able to get footage of Yaoyao in the game.

**Reply by somnisomni** [reply]
> Curious about why not CBT1 specifically?

I did a small, simple-simple personal investigation about how CBT1 communicates with the server (I'm also one who want to play these early beta versions...XD). As I'm not a member of GC discord and I don't know how the one explained this, so I'm going to explain in my way:

At least, the endpoint `(dispatch)/query_region_list`, should return the region list data in **plain JSON format** for CBT1, not protobuf-encoded string or something else used in later versions.

Also in the same endpoint **should not** contain some properties like `clientSecretKey` which also can be found in later versions. (The game will throw an exception if there are some 'unknown' properties in response.)

So it will be a challenging thing to make CBT1 works with the private server, since the version requires existing server logic to be rewritten, I think.

(BTW I really looking forward for any private server gonna support CBT1 and other early test versions in sometime XD)

**Reply by FeroxAmarokk** [reply]
i know its been more than 1 year but i wanted to know if you were able to find more stuff. i would love to experience cbt1. im not knowledgeable in private servers or just servers in general but it would be amazing to play it.

**Reply by 4Benj** [reply]
> i know its been more than 1 year but i wanted to know if you were able to find more stuff. i would love to experience cbt1. im not knowledgeable in private servers or just servers in general but it would be amazing to play it.

*cough cough* [soggy](https://github.com/LDAsuku/soggy) *cough cough*
